api/auth_middleware.py

```
@@ -0,0 +1,133 @@
+            for cred_path in [p for p in candidate_paths if p]:
+                if os.path.exists(cred_path):
+                    cred = credentials.Certificate(cred_path)
+                    firebase_admin.initialize_app(cred)
+                    _firebase_initialized = True
+                    break
+            else:
+                # For development, allow initialization without credentials
+                # In production, this should be required
+                print("Warning: Firebase credentials not found. Authentication will be disabled.")
+                print("Set FIREBASE_CREDENTIALS or FIREBASE_CREDENTIALS_PATH environment variables.")
+except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)